wasmtime-runtime/src/traphandlers.rs

```diff
@@ -115,11 +120,18 @@ fn trap_code_to_expected_string(trap_code: ir::TrapCode) -> String {
 /// return values will be written.
 #[no_mangle]
 pub unsafe extern "C" fn wasmtime_call_trampoline(
-    vmctx: *mut VMContext,
+    callee_vmctx: *mut VMContext,
+    caller_vmctx: *mut VMContext,
```

Comment:
Could we make this an `Option<*mut VMContext>` instead? There are a few calls to this function which pass `ptr::null_mut()`, which isn't as clear to me as passing `None`

Reply:
An `Option<*mut VMContext>` wouldn't be ABI-compatible with a C pointer, but an `Option<NonNull<VMContext>>` is [documented](https://doc.rust-lang.org/std/ptr/struct.NonNull.html) to be pointer-sized, so I'll look into using that.